### DISCUSSION TITLE
Etherpad service cannot be started as “etherpad” user

### DISCUSSION THREAD

**Discussion by stdoma** [ether/etherpad-lite#6540] [Q&A]
Hello!
I would like to update the Etherpad version on my server.
So far the whole thing works quite well.
The instructions describe the whole thing quite well: https://github.com/ether/etherpad-lite/wiki/How-to-deploy-Etherpad-Lite-as-a-service
I cannot run the service as an “etherpad” user. I have to specify “User=root” for it to work. However, I have no idea why this is the case.

**Service configuration:**
`[Unit]
Description=Etherpad-lite, the collaborative editor.
After=syslog.target network.target

[Service]
Type=simple
User=etherpad
Group=etherpad
WorkingDirectory=/home/etherpad/etherpad-lite
Environment=NODE_ENV=production
ExecStart=pnpm run prod
Restart=always

StandardOutput=append:/var/log/etherpad/etherpad.log
StandardError=append:/var/log/etherpad/etherpad-error.log

[Install]
WantedBy=multi-user.target`

**status-informations as etherpad user:**
`× etherpad.service - Etherpad-lite, the collaborative editor.
     Loaded: loaded (/etc/systemd/system/etherpad.service; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Thu 2024-07-25 14:28:36 CEST; 4s ago
   Duration: 2ms
    Process: 20876 ExecStart=pnpm run prod (code=exited, status=203/EXEC)
   Main PID: 20876 (code=exited, status=203/EXEC)
        CPU: 1ms

Jul 25 14:28:36 peertube systemd[1]: etherpad.service: Scheduled restart job, restart counter is at 5.
Jul 25 14:28:36 peertube systemd[1]: Stopped etherpad.service - Etherpad-lite, the collaborative editor..
Jul 25 14:28:36 peertube systemd[1]: etherpad.service: Start request repeated too quickly.
Jul 25 14:28:36 peertube systemd[1]: etherpad.service: Failed with result 'exit-code'.
Jul 25 14:28:36 peertube systemd[1]: Failed to start etherpad.service - Etherpad-lite, the collaborative editor..`

**system**
- Debian: 12
- Node: 18.19
- npm: 9.2
- pnpm: 9.6

The “chown -R” on the work directory was executed normally.
Does anyone have an idea what could be causing this?

**Comment by SamTV12345** [comment]
> Hello! I would like to update the Etherpad version on my server. So far the whole thing works quite well. The instructions describe the whole thing quite well: https://github.com/ether/etherpad-lite/wiki/How-to-deploy-Etherpad-Lite-as-a-service I cannot run the service as an “etherpad” user. I have to specify “User=root” for it to work. However, I have no idea why this is the case.
> 
> **Service configuration:** `[Unit] Description=Etherpad-lite, the collaborative editor. After=syslog.target network.target
> 
> [Service] Type=simple User=etherpad Group=etherpad WorkingDirectory=/home/etherpad/etherpad-lite Environment=NODE_ENV=production ExecStart=pnpm run prod Restart=always
> 
> StandardOutput=append:/var/log/etherpad/etherpad.log StandardError=append:/var/log/etherpad/etherpad-error.log
> 
> [Install] WantedBy=multi-user.target`
> 
> **status-informations as etherpad user:** `× etherpad.service - Etherpad-lite, the collaborative editor. Loaded: loaded (/etc/systemd/system/etherpad.service; enabled; preset: enabled) Active: failed (Result: exit-code) since Thu 2024-07-25 14:28:36 CEST; 4s ago Duration: 2ms Process: 20876 ExecStart=pnpm run prod (code=exited, status=203/EXEC) Main PID: 20876 (code=exited, status=203/EXEC) CPU: 1ms
> 
> Jul 25 14:28:36 peertube systemd[1]: etherpad.service: Scheduled restart job, restart counter is at 5. Jul 25 14:28:36 peertube systemd[1]: Stopped etherpad.service - Etherpad-lite, the collaborative editor.. Jul 25 14:28:36 peertube systemd[1]: etherpad.service: Start request repeated too quickly. Jul 25 14:28:36 peertube systemd[1]: etherpad.service: Failed with result 'exit-code'. Jul 25 14:28:36 peertube systemd[1]: Failed to start etherpad.service - Etherpad-lite, the collaborative editor..`
> 
> **system**
> 
> * Debian: 12
> * Node: 18.19
> * npm: 9.2
> * pnpm: 9.6
> 
> The “chown -R” on the work directory was executed normally. Does anyone have an idea what could be causing this?

Normally this should work. What does the log say? So I'd need the last output of append:/var/log/etherpad/etherpad.log and StandardError=append:/var/log/etherpad/etherpad-error.log for better understanding. I tried it two weeks ago when I repaired a failing instalation of an issuer.

**Comment by chriscroome** [comment]
I had exactly the same issue as described above and the issue was that the log file paths didn't exist, after touching them it was fine.
